### PR TITLE
fix: wait until content visible in notification element

### PR DESCRIPTION
**Pull request by tomivirkki**
## Description

If a `Notification` is being animated open and it contains text, `WebElement.getText()` may still return empty if the notification card hasn't become fully visible yet. This manifests especially with Notifications that are positioned at the top of the screen so that they slide down until visible.

This PR works around the issue by making `getCard()` wait until the notification content is considered visible.

Fixes https://github.com/vaadin/flow-components/issues/4921

## Type of change

- Bugfix

### CODE REVIEW

File: vaadin-notification-flow-parent/vaadin-notification-testbench/src/main/java/com/vaadin/flow/component/notification/testbench/NotificationElement.java [hunk 1]
```diff
@@ -60,6 +62,19 @@ public SearchContext getContext() {
...
+        var card = getPropertyElement("_card");
+
+        try {
+            // Wait until the content becomes visible
+            waitUntil(driver -> {
+                var hasVisbleText = !card.getText().isEmpty();
+                var hasVisibleChildren = card.findElements(By.cssSelector("*"))
+                        .stream().anyMatch(element -> element.isDisplayed());
+                return hasVisbleText || hasVisibleChildren;
+            }, 100);
```
Comment (sissbruecker) [inline vaadin-notification-flow-parent/vaadin-notification-testbench/src/main/java/com/vaadin/flow/component/notification/testbench/NotificationElement.java:74]: The timeout would only occur after 100 seconds which seems excessive for this case (waiting for animation to finish). Did you intend to use 100ms?

File: vaadin-notification-flow-parent/vaadin-notification-testbench/src/main/java/com/vaadin/flow/component/notification/testbench/NotificationElement.java [hunk 2]
```diff
@@ -60,6 +62,19 @@ public SearchContext getContext() {
      * @return the card for the notification
      */
     private TestBenchElement getCard() {
-        return getPropertyElement("_card");
+        var card = getPropertyElement("_card");
+
+        try {
+            // Wait until the content becomes visible
+            waitUntil(driver -> {
+                var hasVisbleText = !card.getText().isEmpty();
```
Comment (sissbruecker) [inline vaadin-notification-flow-parent/vaadin-notification-testbench/src/main/java/com/vaadin/flow/component/notification/testbench/NotificationElement.java:70]:
```suggestion
                var hasVisibleText = !card.getText().isEmpty();
```

File: vaadin-notification-flow-parent/vaadin-notification-testbench/src/main/java/com/vaadin/flow/component/notification/testbench/NotificationElement.java (hunk 1 above)
Comment (tomivirkki) [reply vaadin-notification-flow-parent/vaadin-notification-testbench/src/main/java/com/vaadin/flow/component/notification/testbench/NotificationElement.java]: Good catch. Changed to 1 second

File: vaadin-notification-flow-parent/vaadin-notification-testbench/src/main/java/com/vaadin/flow/component/notification/testbench/NotificationElement.java (hunk 2 above)
Comment (tomivirkki) [reply vaadin-notification-flow-parent/vaadin-notification-testbench/src/main/java/com/vaadin/flow/component/notification/testbench/NotificationElement.java]: Done

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**